python/taichi/lang/field.py

```diff
@@ -250,9 +250,13 @@ def to_torch(self, device=None):
 
     @python_scope
     def from_numpy(self, arr):
-        assert len(self.shape) == len(arr.shape)
-        for i in range(len(self.shape)):
-            assert self.shape[i] == arr.shape[i]
+        if self.shape != arr.shape:
```

Comment:
is it `len(self.shape) == len(arr.shape)`?

Reply:
Yeah, it should be len(self.shape) != len(arr.shape) here. I will update it and add one test case.